api/routes/authRouter.js

```diff
@@ -0,0 +1,115 @@
+ ******************************************* dependencies ******************************************
+ **************************************************************************************************/
+require('dotenv').config();
+const express = require('express');
+const db = require('../../data/helpers/dbHelpers.js');
+const bcrypt = require('bcryptjs');
+const router = express.Router();
+const jwt = require('jsonwebtoken');
+
+/***************************************************************************************************
+ ******************************************** middleware *******************************************
+ **************************************************************************************************/
+function generateToken(user) {
+  const payload = {
+    username: user.username,
+    name: user.name,
+    department: user.department // should come from the database user.department
+  }
+
+  const secret = process.env.JWT_SECRET; // bad practice because of env, it can be hackable
+
+  const options = {
+    expiresIn: 60 // 15 seconds... otherValues(20, '2 days', '10h', '7d'), a number represents seconds (not milliseconds)
+  };
+
+  return jwt.sign(payload, secret, options);
+}
+
+function authenticate(req, res, next) {
+  // the auth token is normally sent in the authorization header
+  const token = req.headers.authorization;
+
+  if (token) {
+    jwt.verify(token, process.env.JWT_SECRET, (err, decodedToken) => {
+      if (err) {
+        res.status(401).json({ message: 'invalid token' });
+      } else {
+        req.decodedToken = decodedToken;
+        next();
+      }
+    })
+  } else {
+    res.status(401).json({ message: 'no token provided' });
+  }
+}
+
+function authorizeDepartments(departments) {
+  return function (req, res, next) {
+    if (departments.includes(req.decodedToken.department)) {
+      next();
+    } else {
+      res.status(403).json({ message: `Access Denied: You must be in one of the following departments [${departments}]` })
+    }
+  }
+}
+
+/***************************************************************************************************
+ ********************************************* Endpoints *******************************************
+ **************************************************************************************************/
+// protect this endpoint so only admin users can see it
+router.get('/users', authenticate, authorizeDepartments([db.settings.departments[0]]), (req, res) => {
+  db.getAllUsers('users')
+    .select('id', 'username', 'department')
+    .then(users => res.status(200).json({ users, decodedToken: req.decodedToken }))
+    .catch(err => res.status(500).json({ error: err }));
+});
+
+// protect this endpoint so only admin and accountant users can see it
+router.get('/accountants/users', authenticate, authorizeDepartments([db.settings.departments[0], db.settings.departments[1]]), (req, res) => {
```

Comment:
You guys all came up w/ your own way to do the departments, I think this way is really clever.